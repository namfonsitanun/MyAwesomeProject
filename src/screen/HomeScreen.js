import React, { Component } from 'react'
import { View, StyleSheet, Text, TextInput, Button, TouchableOpacity } from 'react-native'
import { Count } from '../component/Count.js'

export default class HomeScreen extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      count: 0,
    }
  }

  // componentWillReceiveProps() {
  // }

  handleIncrease = () => {
	  console.log('handleIncrease')
	  this.setState({ count: this.state.count + 1 })
  };

	handleDecrease = () => {
	  console.log('handleIncrease')
	  this.setState({ count: this.state.count - 1 })
	};

	render() {
	  return (
  <View style={styles.container}>
    <Text>Home Screen</Text>
    <Button
      title="Go to Detail"
      onPress={() =>
						this.props.navigation.navigate('Details', {
							count: this.state.count,
						})
					}
    />
    <Count count={this.state.count} />
    <View style={styles.list}>
      <TouchableOpacity id="increase" onPress={this.handleIncrease}>
        <View style={styles.item}>
          <Text>Increase</Text>
        </View>
      </TouchableOpacity>

      <TouchableOpacity id="decrease" onPress={this.handleDecrease}>
        <View style={styles.item}>
          <Text>Decrease</Text>
        </View>
      </TouchableOpacity>
    </View>
  </View>
	  )
	}
}
const styles = StyleSheet.create({
  container: {
    // color: "black",
    flex: 1,
    backgroundColor: 'white',
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    fontSize: 48,
  },
  list: {
    marginTop: 60,
    flexDirection: 'row',
    flexWrap: 'wrap',
    alignItems: 'center',
  },
  item: {
    width: 80,
    height: 80,
    backgroundColor: 'grey',
    margin: 20,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 50,
  },
})
