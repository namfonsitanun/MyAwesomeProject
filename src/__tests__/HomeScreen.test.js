import React from 'react'
import 'react-native'
import HomeScreen from '../screen/HomeScreen.js'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(<HomeScreen />).toJSON()
  expect(tree).toMatchSnapshot()
})
