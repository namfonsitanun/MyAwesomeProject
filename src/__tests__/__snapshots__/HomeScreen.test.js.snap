// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Home Screen
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Go to Detail
      </Text>
    </View>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 48,
          "textAlign": "center",
        }
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginTop": 60,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "skyblue",
            "borderRadius": 50,
            "height": 80,
            "justifyContent": "center",
            "margin": 20,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Increase
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "skyblue",
            "borderRadius": 50,
            "height": 80,
            "justifyContent": "center",
            "margin": 20,
            "width": 80,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Decrease
        </Text>
      </View>
    </View>
  </View>
</View>
`;
